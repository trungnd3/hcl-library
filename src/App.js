import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Layout from './components/Layout/Layout';
import AppRoutes from './components/Route/AppRoutes';

import { selectUser } from './app/action-creators/auth';
import { fetchBorrowedBooks } from './app/action-creators/book';

function App() {
  const user = useSelector(selectUser);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchBorrowedBooks(user));
  }, [dispatch, user]);

  return (
    <Layout>
      <AppRoutes />
    </Layout>
  );
}

export default App;
