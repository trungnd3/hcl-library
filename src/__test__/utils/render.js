// test-utils.jsx
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

// Import your own reducer
import authReducer from '../../app/slices/auth';
import bookReducer from '../../app/slices/book';

function render(
  ui,
  {
    preloadedState,
    store = configureStore({
      reducer: { auth: authReducer, book: bookReducer },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
