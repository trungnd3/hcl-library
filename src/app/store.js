import { configureStore } from '@reduxjs/toolkit';
import authReducer from './slices/auth';
import bookReducer from './slices/book';

export const store = configureStore({
  reducer: {
    auth: authReducer,
    book: bookReducer,
  },
});
