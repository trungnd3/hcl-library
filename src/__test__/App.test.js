import React from 'react';
import App from '../App';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import renderer from 'react-test-renderer';
import { prettyDOM, render, screen, fireEvent } from './utils/render';
import userEvent from '@testing-library/user-event';

let user;
describe('App component using React Testing Library', () => {
  // beforeEach(() => {
  //   render(<App />);
  //   user = userEvent.setup();
  // });

  test('render snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test('has brand name of The Grand ABC Library', () => {
  //   const { container } = screen;
  //   console.log(screen);
  //   const brandName = container.querySelector('.navbar-brand').text;
  //   expect(brandName).toBe('The Grand ABC Library');
  // });

  // test('contains Home navigation', () => {
  //   const { container } = screen;
  //   // console.log(prettyDOM(container.('.nav-link')));
  //   expect(1).toEqual(1);
  // });
});

// describe('test suite of pet peers', () => {
//   let wrapper;
//   beforeEach(() => {
//     wrapper = shallow(
//       <Provider store={store}>
//         <App />
//       </Provider>
//     );
//     console.log(wrapper.debug());
//   });

//   test('renders learn react link', () => {
//     const mounted = mount(
//       <Provider store={store}>
//         <App />
//       </Provider>
//     );
//     // console.log(mounted.debug());
//     // expect(wrapper.find('h1').text()).toContain('Pet');

//     // const { getByText } = render(<App />);
//     // const element = getByText('Pet Peers Portal');
//     // expect(element).toBeInTheDocument();
//   });
// });
