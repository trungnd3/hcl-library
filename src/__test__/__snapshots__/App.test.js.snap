// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component using React Testing Library render snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container"
    >
      <a
        className="active navbar-brand"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        The Grand ABC Library
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="ms-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="active nav-link"
            data-rr-ui-event-key="/"
            disabled={false}
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/login"
            disabled={false}
            href="/login"
            onClick={[Function]}
            style={null}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <main>
    <div
      className="mt-5 text-center"
    >
      <h1>
        Welcome libraian!
      </h1>
    </div>
  </main>,
]
`;
